docs/architecture/Overview.md

```diff
@@ -0,0 +1,164 @@
+
+- An [overview](https://azure.github.io/communication-ui-library/?path=/docs/overview--page) of the main offering.
+- An understanding of the [UI composites](https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-composites--page) and the [adapters](https://azure.github.io/communication-ui-library/?path=/docs/composite-adapters--page) that power them.
+- An understanding of the [UI components](https://azure.github.io/communication-ui-library/?path=/docs/overview-uicomponents--page) and the [stateful data layer](https://azure.github.io/communication-ui-library/?path=/docs/statefulclient-overview--page) that powers them.
+
+These documents are geared towards customers. As such, they focus primarily on getting off the ground using the different APIs available in the library and some intricacies of _how_ to use the API.
+
+Documentation in this directory dives into the _why_.
+
+## Architecture - layering
+
+At a high level, the library exposes two distinct layers of APIs.
+
+- A lower-abstraction API in the form of UI components, stateful ACS modality clients (chat / calling) and bindings between the two. The focus in this layer is in giving the customers maximum flexibility. The API makes the default use-cases easier but not at the cost of making non-default use-cases impossible.
+
+- A higher-abstraction API in the form of UI composites, powered by adapters. The focus in this layer is in addressing specific end-to-end use-cases well. While this layer does provide customers some customizability, it choses to make some non-default use-cases hard or impossible. By restricting the set of supported use-cases, this layer is able to support the covered use-cases more extensively and with minimal effort from Contoso.
+
+The user-facing documentation in [storybook](https://azure.github.io/communication-ui-library) starts with the composite layer, as it's the easiest API to use for new customers. It then progressively reveals the components API as an alternative for power-users.
+
+This document starts with the component layer, as it forms the foundation for the composite layer in addition to being a public API customers can use.
+
+## Component layer
+
+The component layer provides a set of UI components and bindings that allow the components to work with the ACS backend services. The architecture follows [unidirectional data flow](./ComponentDesign.md).
+
+A typical application that uses the component layer API looks as follows:
+
+
+
+        ┌────────────────────────────────────────────────────────┐
+        │                                                        │
+        │  Contoso Application                                   │
+        │   * Add top-level Providers using StatefulClients      │
+        │   * Use UI components in app, with component bindings  │
```

Comment:
Could add one more point here like:

        │   * Add top-level Providers using StatefulClients      │
        │   * Use UI components in app, with component bindings  │
        -│   * Implement customized components with bindings  │-
        
Or find a good place to explain that bindings layer is also a solution to support fully customized component implememnted by contoso